src/neuroglancer/viewer.ts

```diff
@@ -773,31 +812,70 @@ export class Viewer extends RefCounted implements ViewerState {
     }
   }
 
-  postJsonState() {
-    // if jsonStateServer is not present prompt for value and store it in state
-    if (!this.jsonStateServer.value) {
-      this.promptJsonStateServer(
-          'No state server found. Please enter a server URL, or hit OK to use the default server.');
+  loadFromJsonUrl() {
+    var urlParams = new URLSearchParams(window.location.search);
+    if (urlParams.has('json_url')) {
+      let json_url = urlParams.get('json_url')!;
+      history.replaceState(null, '', removeParameterFromUrl(window.location.href, 'json_url'));
+      this.resetStateWhenEmpty = false;
+      StatusMessage
+          .forPromise(
+              cancellableFetchOk(json_url, {}, responseJson).then(response => {
+                this.state.restoreState(response);
+              }),
+              {
+                initialMessage: `Retrieving state from json_url: ${json_url}.`,
+                delay: true,
+                errorPrefix: `Error retrieving state: `,
+              })
+          .finally(() => {
+            this.resetStateWhenEmpty = true;
+          });
     }
+  }
 
+  postJsonState(get?: UrlType) {
```

Comment:
Copied from `SaveDialog`: `get` should be a bit more descriptive, e.g. `getUrlType`